app/src/main/java/de/danoeh/antennapod/activity/AboutActivity.java

```diff
@@ -106,35 +101,32 @@ private void loadAsset(String filename) {
                             "    </style>" +
                             "</head><body><p>" + webViewData + "</p></body></html>";
                     webViewData = webViewData.replace("\n", "<br/>");
-                    depth++;
                 } else {
                     depth = 0;
                 }
                 webViewData = String.format(webViewData, colorString);
-                subscriber.onNext(webViewData);
+                subscriber.onSuccess(webViewData);
             } catch (IOException e) {
+                Log.e(TAG, Log.getStackTraceString(e));
                 subscriber.onError(e);
             } finally {
                 IOUtils.closeQuietly(input);
             }
-            subscriber.onCompleted();
         })
                 .subscribeOn(Schedulers.newThread())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(
-                        webviewData ->
-                                webview.loadDataWithBaseURL("file:///android_asset/", webviewData, "text/html", "utf-8", "about:blank"),
+                        webViewData ->
+                                webView.loadDataWithBaseURL("file:///android_asset/", webViewData.toString(), "text/html", "utf-8", "about:blank"),
                         error -> Log.e(TAG, Log.getStackTraceString(error))
                 );
     }
 
     @Override
     public void onBackPressed() {
-        Log.d(TAG, "depth: " + depth);
-        if(depth == 1) {
-            loadAsset("about.html");
-        } else if(depth > 1) {
-            webview.goBack();
+        if (depth != 0) {
+            webView.goBack();
+            depth--;
```

Comment:
On Android 8 (emulated), this still does not work. When pressing the contributors, you have to press back twice. After the first press, the webview does not load the main about page again (does nothing).

If you use the actual url (`"file:///android_asset/" + webViewData.toString()`) instead of `about:blank` as the history parameter of `loadDataWithBaseURL`, you can remove the depth variable (I think thus is the most elegant solution).